homeassistant/components/glances/coordinator.py

```diff
@@ -0,0 +1,42 @@
+"""Coordinator for Glances integration."""
+from datetime import timedelta
+import logging
+from typing import Any
+
+from glances_api import Glances, exceptions
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GlancesDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Get the latest data from Glances api."""
+
+    def __init__(
+        self, hass: HomeAssistant, config_entry: ConfigEntry, api: Glances
+    ) -> None:
+        """Initialize the Glances data."""
+        self.hass = hass
+        self.config_entry: ConfigEntry = config_entry
```

Comment:
Please move the type annotation to a class attribute named `config_entry` above the init method. We need to override the parent class type annotation of this attribute and its more clear to do it that way.